libs/application/templates/new-primary-school/src/lib/NewPrimarySchoolTemplate.ts

```diff
@@ -73,13 +77,37 @@ const NewPrimarySchoolTemplate: ApplicationTemplate<
           ],
         },
         on: {
-          SUBMIT: States.DRAFT,
+          SUBMIT: [
+            {
+              target: States.DRAFT,
+              cond: (application) => {
+                const { children } = getApplicationExternalData(
+                  application?.application?.externalData,
+                )
+
+                //
+                // No child found
+                if (!children || children.length === 0) {
+                  return false
+                }
+
+                // Check if the applicant has some children at primary school age
+                return children.some((child) =>
+                  isChildAtPrimarySchoolAge(child.nationalId),
+                )
+              },
```

Comment:
Move functionality into `NewPrimarySchoolTemplateUtils.ts`?
```suggestion
              cond: (application) => canApply(),
```

Reply:
Fixed